substrate/primitives/trie/src/recorder.rs

```diff
@@ -379,6 +379,19 @@ impl<H: Hasher, I: DerefMut<Target = RecorderInner<H::Out>>> trie_db::TrieRecord
 				// that the value doesn't exist in the trie.
 				self.update_recorded_keys(full_key, RecordedForKey::Value);
```

Comment:
this should switch to RecordedForKey::None I think

Reply:
No the current value is correct. The comment also says why we pass there `Value`.

Reply:
I don't understand it ([this](https://github.com/paritytech/trie/blob/0b9bede27b33da6395e83ed13c523ed2ae913fc6/trie-db/src/recorder.rs#L74) did make sense to me).

Reply:
Good find :see_no_evil: 

https://github.com/paritytech/trie/pull/202

See this docs: https://github.com/paritytech/trie/blob/692ee41a6bd0df36252663d2f7974ab1c368d9a0/trie-db/src/lib.rs#L193-L220

Maybe I should have given `None` a different name, but the docs are explaining it.